api/agent/agent.go

```diff
@@ -1115,21 +1087,21 @@ func (a *agent) runHotReq(ctx context.Context, call *call, state ContainerState,
 				state.UpdateState(ctx, ContainerStatePaused, call.slots)
 			}
 			continue
-		case <-evictor.C:
+		case <-c.GetEvictChan():
 		}
 		break
 	}
 
-	evictor.SetEvictable(false)
-
 	// if we can acquire token, that means we are here due to
-	// abort/shutdown/timeout, attempt to acquire and terminate,
+	// abort/shutdown/timeout/evict, attempt to acquire and terminate,
 	// otherwise continue processing the request
 	if call.slots.acquireSlot(s) {
 		slot.Close()
 		return false
 	}
 
+	c.SetEvictable(call, false)
```

Comment:
Is it worth adding a note here to remind people that the `SetEvictable(false)` needs to happen after slot acquisition so we re-create a token if the evictor already fired?

Reply:
done